index/postings.go

```diff
@@ -329,42 +327,53 @@ func (it *intersectPostings) At() uint64 {
 
 func (it *intersectPostings) doNext(id uint64) bool {
 	for {
-		if !it.b.Seek(id) {
-			return false
-		}
-		if vb := it.b.At(); vb != id {
-			if !it.a.Seek(vb) {
+		find := true
+		for _, p := range it.arr {
+			if !p.Seek(it.cur) {
 				return false
 			}
-			id = it.a.At()
-			if vb != id {
-				continue
+			if p.At() > it.cur {
+				it.cur = p.At()
+				find = false
```

Comment:
We can directly continue to the outer loop here and skip few `Seek` calls.

Reply:
This doesn't change a lot of things according to the benchmarks, but good to have yes.